Tools/Schema.NET.Tool/SchemaSourceGenerator.cs

```diff
@@ -85,6 +80,57 @@ private static string RenderClass(GeneratorSchemaClass schemaClass)
 
         var parentType = schemaClass.Parents.FirstOrDefault();
 
+        var classModifiers = schemaClass.IsCombined ? " abstract" : string.Empty;
+        var classImplements = parentType is not null ? $" {parentType.Name}," : string.Empty;
+        var allProperties = schemaClass.Properties.OrderBy(x => x.Order).ToArray();
+
+        return
+            $$"""
+            #nullable enable
+            namespace Schema.NET;
+
+            using System;
+            using System.Collections.Generic;
+            using System.Text.Json;
+            using System.Text.Json.Serialization;
+
+            {{RenderInterface(schemaClass, parentType)}}
+
+            /// <summary>
+            /// {{SourceUtility.RenderDoc(4, schemaClass.Description)}}
+            /// </summary>
+            public{{classModifiers}} partial class {{schemaClass.Name}} :{{classImplements}} I{{schemaClass.Name}}, IEquatable<{{schemaClass.Name}}>
+            {
+                /// <summary>
+                /// Gets the name of the type as specified by schema.org.
+                /// </summary>
+                [JsonPropertyName("@type")]
+                [JsonPropertyOrder(1)]
+                public override string Type => "{{schemaClass.Name}}";
+
+            {{SourceUtility.RenderItems(allProperties, RenderClassProperty, 4, SourceDelimeter.NewLineSpace)}}
+
+            {{RenderClassTrySetValue(allProperties, 4)}}
+
+            {{RenderClassTryGetVaueOneOrMany(allProperties, 4)}}
+
+            {{RenderClassTryGetVaueIValues(allProperties, 4)}}
+
+            {{RenderClassEquals(schemaClass, allProperties, 4)}}
```

Comment:
Is it just me or does the indenting look wrong for these? Think it needs to be one further in.

Reply:
Yeah, I think we could refactor this to make it read better. The fact is, these functions indent each line of code they generate by the number of characters supplied. Perhaps it could be better if they did not indent the first line.

Reply:
Ok if we leave this for now? I'm currently renovating my house and have little time to contribute.

Reply:
I'm currently renovating my house and am not going to have time to resolve this issue. @Turnerj Is it ok if we merge this and release what we have for now?

Reply:
Yeah, all good mate 🙂